### PR TITLE
Fix parentheses around the index arg of a non-sugared index operation

### LINKED ISSUE #1454: Bug: "ast changed" with valid code
Ocamlformat version: 0.15.0

**Describe the bug**
Ocamlformat fails to format some valid code.

**How to Reproduce**
Input:
```
let foo = (String.get [@bar]) filename (len - 1) = 'i'
```
Ocamlformat reports :
```
  Please report this bug at https://github.com/ocaml-ppx/ocamlformat/issues.
  BUG: ast changed.

  Code: -32603 
```

Edit: please ignore this part below 😅 
~~Also note something else, if the attribute is removed, ocamlformat succeeds, but the output is missing the `String.get` call:
Input:~~
```
let foo = String.get filename (len - 1) = 'i'
```
~~Output:~~
```
let foo = filename.[len - 1] = 'i'
```
----

The selected options are just `m=120` but can reproduce without that as well.

### PR DESCRIPTION
**Pull request by gpetiot**
fix #1454, no diff with test_branch

### CODE REVIEW

File: lib/Ast.ml [hunk 1]
```diff
@@ -1644,7 +1644,8 @@ end = struct
           match i.[0] with
           | '!' | '?' | '~' -> Some (High, Non)
           | _ -> Some (Apply, Non) ) )
-      | Pexp_apply ({pexp_desc= Pexp_ident ident; _}, args)
+      | Pexp_apply
+          ({pexp_desc= Pexp_ident ident; pexp_attributes= []; _}, args)
         when Option.is_some (Indexing_op.get_sugar ident args) -> (
```
Comment (emillon) [inline lib/Ast.ml:1649]: Just to make sure, is there still a test that covers this branch now?

Comment (gpetiot) [reply lib/Ast.ml:1649]: Yes this is the default case, when there is `String.get s i` that will automatically get sugared into the bracket form, we have some tests already.

Comment (emillon) [reply lib/Ast.ml:1649]: allright!

**Review by emillon — APPROVED** [review]

File: lib/Ast.ml (hunk 1 above)
Comment (Julow) [reply lib/Ast.ml:1649]: Perhaps we should directly pass the first argument of `Pexp_apply` to `get_sugar` and move such details where they make sense ?

Comment (gpetiot) [reply lib/Ast.ml]: Good idea, done!

**Pull request merged**